lib/backend/syncersv1/updateprocessors/rules.go

```diff
@@ -46,6 +49,45 @@ func RuleAPIV2ToBackend(ar apiv2.Rule) model.Rule {
 		notICMPCode = ar.NotICMP.Code
 		notICMPType = ar.NotICMP.Type
 	}
+	
+	// If we have any selector specified, then we may need to add the namespace selector.
+	// We do this if this policy is namespaced AND if the Selector does not have any other
+	// k8s namespace (profile label) selector in it.
+	// TODO this is TEMPORARY CODE:  We currently do a simple regex to search for kns. in the
+	// selector to see if we are performing k8s namespace queries.
+	nsSelector := fmt.Sprintf("%s == '%s'", apiv2.LabelNamespace, ns)
+	if ns != "" && (ar.Source.Selector != "" || ar.Source.NotSelector != "") {
```

Comment:
We're not doing anything with `NotSelector` below. I think we need to namespace that as well (if present) using the same logic as for `Selector` in order to be technically correct.

Reply:
Discussed offline and this is fine.